tables/elb_access_log/elb_access_log_table.go

```diff
@@ -0,0 +1,99 @@
+package elb_access_log
+
+import (
+	"time"
+
+	"github.com/rs/xid"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-aws/sources/s3_bucket"
+	"github.com/turbot/tailpipe-plugin-aws/tables"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/mappers"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const ElbAccessLogTableIdentifier = "aws_elb_access_log"
+
+func init() {
+	// Register the table, with type parameters:
+	// 1. row struct
+	// 2. table config struct
+	// 3. table implementation
+	table.RegisterTable[*ElbAccessLog, *ElbAccessLogTable]()
+}
+
+const elbLogFormat = `$type $timestamp $elb $client $target $request_processing_time $target_processing_time $response_processing_time $elb_status_code $target_status_code $received_bytes $sent_bytes "$request" "$user_agent" $ssl_cipher $ssl_protocol $target_group_arn "$trace_id" "$domain_name" "$chosen_cert_arn" $matched_rule_priority $request_creation_time "$actions_executed" "$redirect_url" "$error_reason" "$target_list" "$target_status_list" "$classification" "$classification_reason" $conn_trace_id`
+const elbLogFormatNoConnTrace = `$type $timestamp $elb $client $target $request_processing_time $target_processing_time $response_processing_time $elb_status_code $target_status_code $received_bytes $sent_bytes "$request" "$user_agent" $ssl_cipher $ssl_protocol $target_group_arn "$trace_id" "$domain_name" "$chosen_cert_arn" $matched_rule_priority $request_creation_time "$actions_executed" "$redirect_url" "$error_reason" "$target_list" "$target_status_list" "$classification" "$classification_reason"`
```

Comment:
When is this format used? Is this something a user can configure?

Reply:
These constants define the expected log format for parsing AWS ALB (Application Load Balancer) access logs. They act as templates used by the `GonxMapper`, a function from the SDK responsible for extracting structured fields from raw log entries.
Users cannot configure these formats at runtime, as they are hardcoded within the plugin to align with AWS ELB log structures. If AWS modifies the log format in the future, these constants may need to be updated in the code.
Since I updated `conn_trace_id` field as `omitempty`, and the `GonxMapper` do not requires every field in the format to be present. Therefore, elbLogFormatNoConnTrace can be removed now. Will update the `NewGonxMapper` to use only elbLogFormat, which includes the conn_trace_id field.